### PR TITLE
Support File E@R in New/Set-AzureRMStorageAccount, and Set Default Parameter Set of Set-AzureStorageAccount to AccountTypeParamSet

### PR DESCRIPTION
**Pull request by blueww**
## Description
The PR include 2 changes:
1. Enable File E@R in SRP Storage Account, this include:
   1) Upgrade SRP SDK to 6.1.0-preview for storage (Add new adapt for it in Common.Storage)
   2) Add File E@R feature in New/Set-AzureRMStorageAccount
   3) As the SRP SDK has some interface change in Storage Usage, and Account Kind, add code to handle it, not interface change to user.

2. Set Default Parameter Set of Set-AzureStorageAccount to AccountTypeParamSet, this is to resolve the issue in https://github.com/Azure/azure-powershell/issues/3116#issuecomment-277214038 
     1) As GeoReplicationEnabled is deprecated, we should set Default Parameter Set to AccountTypeParamSet
     2) As the GeoReplicationEnabled deprecated warning will only show when use GeoReplicationEnabledParamSet, and only Parameter "GeoReplicationEnabled" in this parameter set, so set Default Parameter Set to AccountTypeParamSet can solve the issue.

Test:
1) The storage regression is 100% pass
2) On Demand test Pass: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1391/

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ Y] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [Y ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [ Y] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [Y ] Title of the pull request is clear and informative.
- [Y ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [Y ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [Y ] Pull request includes test coverage for the included changes. -Unit test are added 
- [Y ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [Y ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [ Y] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [Y ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [Y ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [Y ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -1,4 +1,11 @@
-﻿## 2017.01.18 - Version 3.4.0
+﻿## 2017.02.20 - Version 3.5.0
```
Comment (markcowl) [inline ChangeLog.md]: Please do not edit this.  Instead, please edit the changelog in your directory - the changes should go under the 'Current Release' section

**Comment by blueww** [comment]
The change log is updated to use the feature change log.

**Comment by blueww** [comment]
The Unit test is added for Storage new feature.

**Comment by markcowl** [comment]
on demand run (including all changes) here:  http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1392/

LGTM once the build passes.  Executing an on-demand run due to changes to RedisCache tests

**Pull request merged**